firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -226,93 +358,77 @@ public void onClick(DialogInterface dialogInterface, int i) {
     return alertDialog;
   }
 
-  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
-  @NonNull
-  public Task<Void> signInTester() {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInCancellationSource.cancel();
-    }
-
-    signInCancellationSource = new CancellationTokenSource();
-    signInTaskCompletionSource = new TaskCompletionSource<>(signInCancellationSource.getToken());
-
-    Context context = firebaseApp.getApplicationContext();
-    AlertDialog alertDialog = getSignInAlertDialog(context);
-    alertDialog.show();
-
-    return signInTaskCompletionSource.getTask();
-  }
-
-  private void setSignInTaskCompletionError(FirebaseAppDistributionException e) {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInTaskCompletionSource.setException(e);
-    }
+  private OnSuccessListener<String> getFidGenerationOnSuccessListener(Context context) {
+    return new OnSuccessListener<String>() {
+      @Override
+      public void onSuccess(String fid) {
+        Uri uri =
+            Uri.parse(
+                String.format(
```

Comment:
Since this is de facto an API call, consider moving into the FirebaseAppDistroApiClient as a static method

Reply:
We use this Uri for redirecting to the Alkali signin flow, and use it to start the redirection intent. I think it should belong here and not the TesterApiClient since it is not related to the TesterApi